packages/teams-js/src/public/media.ts

```diff
@@ -350,12 +367,125 @@ export namespace media {
    */
   export interface AudioProps {
     /**
-     * Optional; the maximum duration in minutes after which the recording should terminate automatically.
+     * Optional; the maximum duration in minutes after which the recording should terminate automatically
      * Default value is defined by the platform serving the API.
      */
     maxDuration?: number;
   }
 
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * Base class which holds the callback and notifies events to the host client
+   */
+  abstract class MediaController<T> {
+    protected controllerCallback: T;
+
+    public constructor(controllerCallback?: T) {
+      this.controllerCallback = controllerCallback;
+    }
+
+    protected abstract getMediaType(): MediaType;
+
+    /**
+     * @hidden
+     * Hide from docs
+     * --------
+     * This function will be implemented by the respective media class which holds the logic
+     * of specific events that needs to be notified to the app.
+     * @param mediaEvent indicates the event signed by the host client to the app
+     */
+    protected abstract notifyEventToApp(mediaEvent: MediaControllerEvent): void;
+
+    /**
+     * @hidden
+     * Hide from docs
+     * --------
+     * Function to notify the host client to programatically control the experience
+     * @param mediaEvent indicates what the event that needs to be signaled to the host client
+     */
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent): Promise<void> {
```

Comment:
To have a backward compatibility make it an overloaded function. You can see the selectMedia() method for reference.
The implementation would look something like below
```
   protected notifyEventToHost(mediaEvent: MediaControllerEvent): Promise<void>;
    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): void;
    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): Promise<void> {
      ensureInitialized(FrameContexts.content, FrameContexts.task);
      try {
        isApiSupportedOnMobile(nonFullScreenVideoModeAPISupportVersion);
      } catch (err) {
        const wrappedFunction: InputFunction<void> = () => Promise.reject(err);
        return callCallbackWithSdkErrorFromPromiseAndReturnPromise(wrappedFunction, callback);
      }
      const mediaControllerParams: MediaControllerParam = {
        mediaType: this.getMediaType(),
        mediaControllerEvent: mediaEvent,
      };
      const wrappedFunction: InputFunction<void> = () =>
        Promise.resolve(sendAndHandleSdkError('media.controller', mediaControllerParams));
      return callCallbackWithSdkErrorFromPromiseAndReturnPromise(wrappedFunction, callback);
    }
```

Reply:
@aeunms How much work is it to add this overload? If it's easy and simple we can do it as part of the merge, if not we can open a bug and assign it out.

Reply:
I missed resolving this, it has already been overloaded.

Reply:
doing a double take- it looks like actually, I just missed overloading this. Will look at this with fresher eyes.

Reply:
As discussed offline, the promisifying (and subsequently, the overloading) will be done by Niharika here. I have created a bug item and will link to it in our Teams git magic thread.